pkg/kube/client.go

```diff
@@ -628,3 +645,66 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
 
 	return err
 }
+
+//get an kubernetes resources's relation pods
+// kubernetes resource used select labels to relate pods
+func (c *Client) getSelectRelationPod(info *resource.Info, objPods map[string][]api.Pod) (map[string][]api.Pod, error) {
+	if info == nil {
+		return objPods, nil
+	}
+
+	err := info.Get()
+	if err != nil {
+		return objPods, err
+	}
+
+	c.Log("get relation pod of object: %s/%s/%s", info.Namespace, info.Mapping.GroupVersionKind.Kind, info.Name)
+
+	versioned, err := c.AsVersionedObject(info.Object)
+	if runtime.IsNotRegisteredError(err) {
+		return objPods, nil
+	}
+	if err != nil {
+		return objPods, err
+	}
+
+	selector, err := getSelectorFromObject(versioned)
```

Comment:
We don't have to worry about this error if we do a check like [this one](https://github.com/yan234280533/helm/compare/patch-pod...thomastaylor312:patch-pod#diff-a483d6c0863082c3df21f4aad513afe2R674). We do need to do that check in the link because otherwise we end up with an empty selector and that will select all pods.

Reply:
I modify the exception check of select string.